core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -2794,42 +2797,59 @@ protected String preparePassword(String password, String saltValue) throws UserS
     protected String preparePassword(Object password, String saltValue) throws UserStoreException {
 
         Secret credentialObj;
+        String passwordHash = new String();
         try {
             credentialObj = Secret.getSecret(password);
         } catch (UnsupportedSecretTypeException e) {
             throw new UserStoreException("Unsupported credential type", e);
         }
+        String digestFunction = realmConfig.getUserStoreProperties().get(JDBCRealmConstants.DIGEST_FUNCTION);
+        if (digestFunction != null) {
 
-        try {
-            String passwordString;
-            if (saltValue != null) {
-                credentialObj.addChars(saltValue.toCharArray());
-            }
+            if (digestFunction.equals(UserCoreConstants.RealmConfig.PASSWORD_HASH_METHOD_PLAIN_TEXT)) {
+                passwordHash = new String(credentialObj.getChars());
+                return passwordHash;
+            } else if (digestFunction.equals("SHA-256")) {
+                try {
+                    if (saltValue != null) {
+                        credentialObj.addChars(saltValue.toCharArray());
+                    }
+                    MessageDigest digest = MessageDigest.getInstance(digestFunction);
+                    byte[] byteValue = digest.digest(credentialObj.getBytes());
+                    passwordHash = Base64.encode(byteValue);
 
-            String digestFunction = realmConfig.getUserStoreProperties().get(JDBCRealmConstants.DIGEST_FUNCTION);
-            if (digestFunction != null) {
-                if (digestFunction.equals(UserCoreConstants.RealmConfig.PASSWORD_HASH_METHOD_PLAIN_TEXT)) {
-                    passwordString = new String(credentialObj.getChars());
-                    return passwordString;
+                } catch (NoSuchAlgorithmException e) {
+                    String msg = "Error occurred while preparing password.";
+                    if (log.isDebugEnabled()) {
+                        log.debug(msg, e);
+                    }
+                    throw new UserStoreException(msg, e);
                 }
 
-                MessageDigest digest = MessageDigest.getInstance(digestFunction);
-                byte[] byteValue = digest.digest(credentialObj.getBytes());
-                passwordString = Base64.encode(byteValue);
-            } else {
-                passwordString = new String(credentialObj.getChars());
-            }
+            } else if (digestFunction.equals("PBKDF2WithHmacSHA256")) {
```

Comment:
We have decided to make this only `PBKDF2` right and introduce another attribute to configure hashing func right?